server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -121,13 +121,17 @@ static LogicalPlan buildJoinTree(List<AnalyzedRelation> from,
 
         boolean isFiltered = validWhereConditions.symbolType().isValueSymbol() == false;
 
+        LogicalPlan newLhs = plan.apply(lhs);
+        LogicalPlan newRhs = plan.apply(rhs);
         LogicalPlan joinPlan = new JoinPlan(
-            plan.apply(lhs),
-            plan.apply(rhs),
+            AbstractJoinPlan.buildOutputs(newLhs.outputs(), newRhs.outputs(), joinType),
```

Comment:
The outputs from lhs/rhs are already adapted by the split points therefore only necessary outputs are included.